src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -237,7 +237,7 @@ export default class CanvasController {
         const alpha = extrapolate_range_clamp(0.1, elapsed, 0.4, 0, 1);
         const framestep = Math.round(extrapolate_range_clamp(1, GameController.game.speedup, 10, 30, 1));
 
-        if (this.canvas.dirty || (!GameController.game_paused() && prop.time.frames % framestep === 0) || elapsed < 1) {
+        if (this.canvas.dirty || (!GameController.game_paused() && prop.time.frames % framestep === 0) || elapsed > GameController.getGameOption(GAME_OPTION_NAMES.RADAR_UPDATE_DELAY)) {
```

Comment:
oh my this is some tricky logic.  This was here before, so I know it's not your fault.  

at the very least, could you pull some of this out to `const`s before the if block?
```javascript
const updateDelay = GameController.getGameOption(GAME_OPTION_NAMES.RADAR_UPDATE_DELAY);
const shouldUpdate = this.canvas.dirty || !GameController.game_paused() && prop.time.frames % framestep === 0;

if (shouldUpdate || elapsed > updateDelay) {
  // ...
}
```

I still don't like it, but at least its easier to read.  The `CanvasController` will get a good fixing some day in the future.